### PR TITLE
Audio player blah blah blah

### PR DESCRIPTION
**Pull request by ljewalsh**

### CODE REVIEW

File: src/journal/views.py [hunk 1]
```diff
@@ -76,6 +76,11 @@ def single_issue(request, issue_number):
     cover = Cover.objects.filter(issue=requested_issue).first()
     story_set = requested_issue.get_story_set().order_by('number')
 
+    height_of_bandcamp_plugin = 120
```
Comment (metamorfoso) [inline src/journal/views.py]: We should be able to do this in the templates -- we're passing `story_set` as part of the context anyway, so the markup and styling logic can be worked out in jinja.

Ideally we'd have logic that assigns html attributes (like `id` or `class`), and then css/sass that actually applies height values.

Ping me if that doesn't make sense.

Comment (ljewalsh) [reply src/journal/views.py]: Good point!

**Comment by ljewalsh** [comment]
Have moved the code to the template. Fixed some css too. 

Problem is, the bandcamp plugin on the issue page pushes the content down too far and you can't see all the stories in an issue. I'll attach a screenshot. Not sure what the best solution is

**Comment by ljewalsh** [comment]
Ok I added the audio/contents toggle. It looks WAY better. I haven't really styled the toggle buttons that much. They look a bit weird how they are. Thought I'd leave that for a discussion

File: resources/c/partial/tab.sass [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+.inactive
+  opacity: .6
+
+
+.tabcontent
+  float: left
+  padding: 0px 12px
+  width: 70%
+  border-left: none
+  height: 300px
```
Comment (metamorfoso) [inline resources/c/partial/tab.sass]: I'm not sure how you're getting this stylesheet file to apply... It needs to be imported in `application.sass` to take effect.

File: templates/journal/single_issue.html [hunk 3]
```diff
@@ -4,104 +4,113 @@
...
-            $("#cover-info").css("display","inline-block");
-
-        }
+if(PDFObject.supportsPDFs === false){
+   $("#issue-pdf").css("display","none");
+}
 
-        function hideCoverInfo(){
-            $("#cover-info").css("display", "none");
+        PDFObject.embed("{{issue.pdf.url}}", "#issue-pdf", {pdfOpenParams: { view: 'FitW', scrollbar: '1', toolbar: '0', statusbar: '1', messages: '0', navpanes: '1' }});
```
Comment (metamorfoso) [inline templates/journal/single_issue.html]: At the moment this line breaks the site on this page if there is no PDF file uploaded for this issue. The script block needs to check whether there is a pdf file:
`{% if issue.pdf %}<script>...</script>{% endif %}`

File: templates/journal/story.html [hunk 4]
```diff
@@ -7,14 +7,35 @@
...
+            {% set height_of_bandcamp_plugin = [142] %}
+                {% for story in stories_in_issue %}
+                    {% if story.bandcamp_track_id %}
+                        {% if height_of_bandcamp_plugin.append(height_of_bandcamp_plugin.pop() + 33) %}
+                        {% endif %}
+                    {% endif %}
+                {% endfor %}
+            {% set height_of_bandcamp_plugin = height_of_bandcamp_plugin.pop()| string + 'px' %}
+            <div style="height:calc({{height_of_bandcamp_plugin}} + 1rem + 4px)" class="tab">
+                <button class="active" id="contents-tab" onclick="openTab('contents')"><span>Contents</span></button>
```
Comment (metamorfoso) [inline templates/journal/story.html]: This is good, but the text is hard to centre and to make look good. I reckon icons could look better.

**Comment by metamorfoso** [comment]
This wasn't working for me until I removed this section of `mobile.sass`:

```
div.tab
      display: none

  #mobile
      display: block
      margin: 1rem auto

  #desktop:
      display: none

  .mobile-audio-player
    position: relative
    padding-top: 1rem
```

I've made a PR for a branch that works for me locally, and addresses the things I've commented on above. I'll leave it here for you to take or leave as you see fit: this is it https://github.com/metamorfoso/scifi-journal/pull/21

**Comment by metamorfoso** [comment]
I think this is ready to go.

**Pull request merged**